user-console/src/main/java/org/pentaho/mantle/client/ui/BurgerMenuBar.java

```diff
@@ -0,0 +1,320 @@
+    }
+  }
+
+  private void onItemOver( MenuItem item, boolean isOver ) {
+    // Still need to select and unselect items.
+    if ( item != null && item.isEnabled() ) {
+      selectItem( isOver ? item : null );
+      focus();
+    }
+  }
+
+  protected void onEscape() {
+    closeLocal( true );
+  }
+
+  protected void onEnter() {
+    if ( !selectFirstItemIfNoneSelectedLocal() ) {
+      doItemActionLocal( getSelectedItem(), true, true );
+    }
+  }
+
+  protected void ensureVisible( MenuItem menuItem ) {
+    ElementUtils.scrollVerticallyIntoView( menuItem.getElement() );
+  }
+
+  public List<UIObject> getAllItems() {
+    return MenuBarUtils.getAllItems( this );
+  }
+
+  @Override
+  public List<MenuItem> getItems() {
+    return super.getItems();
+  }
+
+  // region Back Menu Items
+
+  /**
+   * Adds a "back" menu item to each descendant menu bar.
+   * <p>
+   *   Should be called on the root of the subtree, once it's fully built,
+   *   or when there have been modifications to it.
+   * </p>
+   *
+   * @see BurgerBackMenuItem
+   */
+  public void addBackItemToDescendantMenus() {
+    addBackItems( null );
+  }
+
+  private void addBackItems( MenuItem parentMenuItem ) {
+    if ( parentMenuItem != null && hasItemsButNoBackItem() ) {
+      // Add a back menu item, followed by a menu item separator.
+      insertSeparator( new PentahoMenuSeparator(), 0 );
+      insertItem( new BurgerBackMenuItem( parentMenuItem.getHTML(), true ), 0 );
+    }
+
+    for ( MenuItem menuItem : getItems() ) {
+      BurgerMenuBar subMenu = (BurgerMenuBar) menuItem.getSubMenu();
+      if ( subMenu != null ) {
+        subMenu.addBackItems( menuItem );
+      }
+    }
+  }
+
+  private boolean hasItemsButNoBackItem() {
+    List<MenuItem> menuItems = getItems();
+    if ( !menuItems.isEmpty() ) {
+      return !( menuItems.get( 0 ) instanceof BurgerBackMenuItem );
+    } else {
+      return false;
+    }
+  }
+  // endregion Back Menu Items
+
+  protected void closeRoot() {
+    PopupPanel menuPopup = getRootPopup();
+    if ( menuPopup != null ) {
+      menuPopup.hide();
+    }
+  }
+
+  protected BurgerMenuBar getRootMenu() {
+    BurgerMenuBar parentMenu = getMenuBarParent();
+    if ( parentMenu != null ) {
+      return parentMenu.getRootMenu();
+    }
+
+    return this;
+  }
+
+  protected PopupPanel getRootPopup() {
+    Widget parent = getRootMenu().getParent();
+    while ( parent != null ) {
+      if ( parent instanceof PopupPanel ) {
+        return (BurgerMenuPopup) parent;
+      }
+
+      parent = parent.getParent();
+    }
+
+    return null;
+  }
+
+  // Local version of private super.findItem( Element ).
+  protected MenuItem findItemLocal( com.google.gwt.dom.client.Element hItem ) {
+    for ( MenuItem item : getItems() ) {
+      if ( item.getElement().isOrHasChild( hItem ) ) {
+        return item;
+      }
+    }
+
+    return null;
+  }
+
+  // Local version of private super.eatEvent( Event ).
```

Comment:
Exact copy of super class MenuBar#eatEvent(Event).  Can we just call the super class function or explain why you have a "local" version instead of just using super class implementation in javadoc?

https://github.com/gwtproject/gwt/blob/release/2.9.0/user/src/com/google/gwt/user/client/ui/MenuBar.java#L1157-L1160